### PR TITLE
[SMALLFIX] Minor Mesos-related fixes

### PR DESCRIPTION
**Pull request by aaudiber**

### CODE REVIEW

File: bin/alluxio-mount.sh [hunk 1]
```diff
@@ -208,8 +208,8 @@ function mount_local() {
       DECL_INIT=$(declare -f init_env)
       DECL_MEM_SIZE_TO_BYTES=$(declare -f mem_size_to_bytes)
       DECL_MOUNT_LINUX=$(declare -f mount_ramfs_linux)
-      sudo bash -O extglob -c "BIN=${BIN}; ${DECL_INIT}; ${DECL_MEM_SIZE_TO_BYTES}; \
```
Comment (aaudiber) [inline bin/alluxio-mount.sh]: Need to pass `ALLUXIO_CONF_DIR` to `mount_ramfs_linux`, which will source `${ALLUXIO_CONF_DIR}/alluxio-env.sh`

File: dev/scripts/generate-tarball.sh [hunk 2]
```diff
@@ -33,14 +33,17 @@ mkdir -p ${WORK_DIR}
...
 rsync -aq --exclude='logs' --exclude='dev' ${HOME} ${REPO_COPY}
 
 pushd ${REPO_COPY} > /dev/null
 git clean -fdx
+git reset --hard HEAD
 BUILD_LOG="${HOME}/logs/build.log"
 echo "Running build and logging to ${BUILD_LOG}"
 mvn -T 4C clean install -Dmaven.javadoc.skip=true -DskipTests -Dlicense.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true -Pmesos ${BUILD_OPTS} | tee ${BUILD_LOG} 2>&1
-VERSION=$(${HOME}/bin/alluxio version)
+touch conf/alluxio-env.sh
```
Comment (aaudiber) [inline dev/scripts/generate-tarball.sh]: `bin/alluxio version` sources `conf/alluxio-env.sh`

File: examples/src/main/java/alluxio/cli/AlluxioFrameworkIntegrationTest.java [hunk 3]
```diff
@@ -127,7 +127,7 @@ public Boolean apply(Void input) {
   private void startAlluxioFramework(Map<String, String> extraEnv) {
     String startScript = PathUtils.concatPath(Configuration.get(PropertyKey.HOME),
         "integration", "bin", "alluxio-mesos.sh");
-    ProcessBuilder pb = new ProcessBuilder(startScript, mMesosAddress, "-w");
+    ProcessBuilder pb = new ProcessBuilder(startScript, mMesosAddress);
```
Comment (aaudiber) [inline examples/src/main/java/alluxio/cli/AlluxioFrameworkIntegrationTest.java]: this flag was doing nothing previously - it needs to come before the other arguments to be noticed

File: integration/bin/alluxio-env-mesos.sh [hunk 4]
```diff
@@ -10,6 +10,9 @@
 # See the NOTICE file distributed with this work for information regarding copyright ownership.
 #
 
+export ALLUXIO_CONF_DIR="${MESOS_SANDBOX}/conf"
```
Comment (aaudiber) [inline integration/bin/alluxio-env-mesos.sh]: moved here from `AlluxioScheduler`

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioWorkerExecutor.java [hunk 5]
```diff
@@ -76,7 +76,7 @@ public void run() {
               UnderFileSystemRegistry.class.getClassLoader());
 
           // TODO(jiri): Consider handling Format.main() failures gracefully.
-          Format.main(new String[] {"master"});
+          Format.main(new String[] {"worker"});
```
Comment (aaudiber) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioWorkerExecutor.java]: @jsimsa was this intentional before?

File: bin/alluxio-mount.sh [hunk 6]
```diff
@@ -208,8 +208,8 @@ function mount_local() {
...
-       ${DECL_MOUNT_LINUX}; mount_ramfs_linux $0"
+      sudo bash -O extglob -c "ALLUXIO_CONF_DIR=${ALLUXIO_CONF_DIR}; BIN=${BIN}; ${DECL_INIT}; \
+${DECL_MEM_SIZE_TO_BYTES}; ${DECL_MOUNT_LINUX}; echo ${ALLUXIO_CONF_DIR}; mount_ramfs_linux $0"
```
Comment (jsimsa) [inline bin/alluxio-mount.sh]: do you still need the `echo` here?

File: dev/scripts/generate-tarball.sh [hunk 7]
```diff
@@ -33,14 +33,17 @@ mkdir -p ${WORK_DIR}
 HOME="${THIS}/../.."
```
Comment (jsimsa) [inline dev/scripts/generate-tarball.sh]: this should really be a Go script ;-)

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioWorkerExecutor.java (hunk 5 above)
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioWorkerExecutor.java]: no, I think that was a copy and paste bug

File: dev/scripts/generate-tarball.sh (hunk 7 above)
Comment (aaudiber) [inline dev/scripts/generate-tarball.sh]: but everyone has bash!

File: bin/alluxio-mount.sh (hunk 6 above)
Comment (aaudiber) [inline bin/alluxio-mount.sh]: removed, thanks

**Comment by aaudiber** [comment]
@jsimsa Comments addressed, PTAL

File: dev/scripts/generate-tarball.sh (hunk 7 above)
Comment (jsimsa) [inline dev/scripts/generate-tarball.sh]: I actually think that Go is more portable than bash (running bash on Windows requires jumping through hoops)

portability aside, Go has additional benefits of being compiled and more importantly there seems to more consensus around what an API of a Go script is (command-line flags +  parameters) vs. shell script (environment variables, command-line arguments, parameters, and in some cases people assume it is okay to sed a bash script in which case every line of bash ends up being the API) ... the smaller the API surface, the easier it generally is to maintain and evolve the implementation

I could go on ... :-)

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**